### ISSUE TITLE
Project system is broken for portable projects referencing other portable projects on VS "15" Preview 3

### ISSUE THREAD

**Issue by KevinH-MS** [dotnet/roslyn#12491]
I have a portable project that references another portable project.  Both of the project.json files contain this:

```
{
  "supports": { },
  "dependencies": {
    "Microsoft.NETCore.Portable.Compatibility": "1.0.1",
    "NETStandard.Library": "1.6.0",
    "System.Collections.Immutable": "1.2.0"
  },
  "frameworks": {
    "netstandard1.3": { }
  }
}
```

When opening the project, I see the following warnings in the Error List:
```
Warning		Unable to add a reference to project 'Runtime'. The targets of Portable Library project 'Runtime' are not the same or compatible with the targets of the current Portable Library project.

A Portable Library project's targets can be changed via the Library tab in the project's properties.	TestImpact.Contracts
```

Eventually, these result in errors (presumably because the references aren't being added to the Compilation's used for IntelliSense).

The issue seems to be that the Path format returned by NewtonSoft is not what the Project Json Reader in the project system expects:

![reader](https://cloud.githubusercontent.com/assets/6464209/16814571/49e546a0-48eb-11e6-8a07-7742e9eb29e0.png)

**Comment by KevinH-MS** [comment]
FYI @ManishJayaswal this is the issue I was talking about in standup where I get tons of spurious errors in the IDE.

**Comment by ManishJayaswal** [comment]
@rchande please let us know when you have a fix. We would like to try out the privates.

**Comment by rchande** [comment]
@KevinH-MS What build of VS were you on?

**Comment by KevinH-MS** [comment]
15.0.25428.1 PREVIEW

**Comment by rchande** [comment]
@KevinH-MS Can you try my a private build of MS.VS.ProjectFlavoring.dll? I've made it available at `\\mlangfs1\public\rchande\projectflavoring`.

Let me know if it fixes the problem. Thanks!

**Comment by KevinH-MS** [comment]
I'm still seeing warnings.  Let's sync up this afternoon and try to debug.

**Comment by KevinH-MS** [comment]
It looks like the warnings were just the array syntax problem.  As soon as we removed those, things seem to work.  Thanks!

**Comment by Pilchie** [comment]
Sounds like this should be closed now?

**Comment by KevinH-MS** [comment]
I haven't verified bits with the array syntax fixed, but the major issue is resolved, yes.
